app/uk/gov/hmrc/tai/service/CodingComponentService.scala

```diff
@@ -33,13 +33,7 @@ class CodingComponentService @Inject()(
   taxFreeAmountComparisonConnector: TaxFreeAmountComparisonConnector) {
 
   def taxFreeAmountComponents(nino: Nino, year: TaxYear)(implicit hc: HeaderCarrier): Future[Seq[CodingComponent]] =
-    taxAccountConnector.codingComponents(nino, year) map {
-      case TaiSuccessResponseWithPayload(codingComponents: Seq[CodingComponent]) =>
-        filterOutZeroAmountsComponents(codingComponents)
-      case TaiNotFoundResponse(_)          => Seq.empty
-      case TaiTaxAccountFailureResponse(e) => throw new RuntimeException(e)
-      case _                               => throw new RuntimeException("could not fetch coding components")
-    }
+    taxAccountConnector.codingComponents(nino, year) map filterOutZeroAmountsComponents
```

Comment:
personally i prefer `.map(f)` and i would use the infix notation for things like boolean logic `x contains y` rather than `x.contains(y)`, etc

Reply:
Thanks, I'll make that change and keep it in mind in future